enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,549 @@
+  deployments that are not isolating the management workloads.
+* We have a goal of running on 4 hyperthreads today, but we do not
+  know what constraints we might be given in the future so we need a
+  solution that is not tied to the current known limit.
+* We want a general approach, that can be applied to all OpenShift
+  control plane components.
+
+### Non-Goals
+
+* This enhancement is focused on CPU resources. Other compressible
+  resource types may need to be managed in the future, and those are
+  likely to need different approaches.
+* This enhancement does not address non-compressible resource
+  requests, such as for memory.
+* This enhancement does not address ways to disable operators or
+  operands entirely.
+* Although the results of this enhancement may be useful for
+  single-node developer deployments, multi-node production
+  deployments, and Internet-of-things devices, those use cases are not
+  addressed directly in this document.
+* This enhancement does not address reducing actual utilization. There
+  is no expectation that a cluster configured to use a small number of
+  cores for management services would offer exactly the same
+  performance as the default. It must be stable and continue to
+  operate reliably, but may respond more slowly.
+* This enhancement assumes that the configuration of management
+  services is done as part of installing the cluster, and cannot be
+  changed later. Future enhancements may address enabling or
+  reconfiguring the feature described here on an existing cluster.
+* This enhancement describes partitioning concepts that could be
+  expanded to be used for other purposes. Use cases for partitioning
+  workloads for other purposes may be addressed by future
+  enhancements.
+
+## Proposal
+
+[Previous attempts](https://github.com/openshift/enhancements/pull/628)
+to solve this problem focused on reducing or scaling requests so that
+the normal scheduling criteria could be used to safely place them on
+CPUs in the shared pool. This proposal reframes the problem, so that
+instead of considering "scaling" or "tuning" the requests we think about
+"isolating" or "partitioning" the workloads away from the non-management
+workloads. This view of the problem is more consistent with how the
+requirement was originally presented by the customer.
+
+We want to define "management workloads" in a flexible way. For the
+purposes of this document, "management workloads" include all
+OpenShift core components necessary to run the cluster, any add-on
+operators necessary to make up the "platform" as defined by telco
+customers, and operators or other components from third-party vendors
+that the customer deems as management rather than operational. It is
+important to note that not all of those components will be delivered
+as part of the OpenShift payload and some may be written by the
+customer or by vendors who are not our partners.
+
+The basic proposal is to provide a way to identify management
```

Comment:
it may be useful to define the 3 pools like follows:

management: cpuset that restricts management components
shared: management + !(dedicated)
dedicated: exclusive

the management pools is not exclusive, but its restricted from the outset to not use the full shared pool whose size may fluctuate.  this also keeps the capability compatible with the cpu manager component.